### ISSUE TITLE
[BUG/FEAT] Operators (mult/div/sub) raise AttributeError on weighted histograms 

### ISSUE THREAD

**Issue by jonas-eschle** [scikit-hep/boost-histogram#693]
**Describe the bug**

Dividing/multiplying/subtracting histograms with weights results in an `AttributeError`, only the addition works.

Is this not implemented yet or is this ill-defined how to propagate the errors?

If the latter, is it worth to have a different error message than an `AttributeError`?

**Steps to reproduce**

Tested on newest release and on dev branch.

```
import boost_histogram as bh

h = bh.Histogram(bh.axis.Regular(10, -5, 5), storage=bh.storage.Weight())
h.fill([1.5, 2.5], weight=[0.5, 1.5])
h + h  # works
h / h  # this errors
h * h  # this errors
h - h  # this errors
```
Error message:
```
AttributeError: 'boost_histogram._core.hist.any_weight' object has no attribute '__imul__'
```
Full traceback

<details>
<summary>Details</summary>

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-16-a9e928048076> in <module>
----> 1 h * h

~/anaconda3/envs/zfit39/lib/python3.9/site-packages/boost_histogram/_internal/hist.py in __mul__(self, other)
    360     ) -> H:
    361         result = self.copy(deep=False)
--> 362         return result._compute_inplace_op("__imul__", other)
    363 
    364     def __rmul__(

~/anaconda3/envs/zfit39/lib/python3.9/site-packages/boost_histogram/_internal/hist.py in _compute_inplace_op(self, name, other)
    399         # Also takes CppHistogram, but that confuses mypy because it's hard to pick out
    400         if isinstance(other, Histogram):
--> 401             getattr(self._hist, name)(other._hist)
    402         elif isinstance(other, tuple(_histograms)):
    403             getattr(self._hist, name)(other)

AttributeError: 'boost_histogram._core.hist.any_weight' object has no attribute '__imul__'

</details>

**Comment by henryiii** [comment]
https://github.com/boostorg/histogram/pull/351 is adding this for division to Boost.Histogram. We try to match Boost.Histogram's support here, so division will be added once that is merged.

**Comment by henryiii** [comment]
And the tracking issue for the rest of them is here: https://github.com/boostorg/histogram/issues/352

We'll follow those as they go in, and include them here.

**Comment by masonproffitt** [comment]
Division is in boostorg/histogram now, so it can get included here, right?

**Comment by henryiii** [comment]
Yep, if it's in Boost.Histogram, it can be added to boost-histogram.

**Comment by JMulder99** [comment]
What is the status of this issue? In the `[Hist](https://github.com/scikit-hep/hist)` project, someone opened an [issue](https://github.com/scikit-hep/hist/issues/427) regarding adding weighted histograms, and another [issue](https://github.com/scikit-hep/hist/issues/463) is regarding using the `.plot_ratio` for weighted histograms.

I think all these issues are caused by the lack of the following or similar attributes:
`AttributeError: 'boost_histogram._core.hist.any_weight' object has no attribute '__itruediv__'`
or
`AttributeError: 'boost_histogram._core.hist.any_weight' object has no attribute '__isub__'`

**Comment by wiso** [comment]
Also subtraction. ROOT supports it
